android/beagle/src/main/java/br/com/zup/beagle/android/builder/action/NavigateBuilder.kt

```diff
@@ -0,0 +1,250 @@
+/*
+ * Copyright 2020 ZUP IT SERVICOS EM TECNOLOGIA E INOVACAO SA
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package br.com.zup.beagle.android.builder.action
+
+import br.com.zup.beagle.android.action.Navigate
+import br.com.zup.beagle.android.action.Route
+import br.com.zup.beagle.android.builder.layout.ScreenComponentBuilder
+import br.com.zup.beagle.android.components.layout.Screen
+import br.com.zup.beagle.builder.BeagleBuilder
+import br.com.zup.beagle.builder.BeagleMapBuilder
+import kotlin.properties.Delegates
+
+fun route(block: RouteBuilder.() -> Unit) = RouteBuilder().apply(block).build()
```

Comment:
What do you think about create some other functions called  `remoteRoute` and `localRoute`?
```suggestion
fun route(block: RouteBuilder.() -> Unit) = RouteBuilder().apply(block).build()
fun remoteRoute(block: RouteRemoteBuilder.() -> Unit) = RouteRemoteBuilder().apply(block).build()
fun localRoute(block: RouteLocalBuilder.() -> Unit) RouteLocalBuilder().apply(block).build()
```
This will help the developer create a route cleaner.

Reply:
I like the idea. Do you mind if I name it in order of the hierarchy? i.e: routeRemote, routeLocal, imagePathRemote, imagePathLocal, etc